# Stock X Data Contest for the data nerds

Your submission should consist of a single pdf file with the following items:

1 chart or graph displaying your results. Do not exceed 1 chart
A short write-up describing your analysis/results. Do not exceed 200 words
One entry per participant
Make sure the pdf has your name and email address somewhere on it

![StockX image](https://s3.amazonaws.com/stockx-sneaker-analysis/wp-content/uploads/2019/02/StockX_Data-Contest_Logo1-1100x1118.jpg "stockX logo")

In [10]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from scipy import stats
import plotly.plotly as py
import plotly.graph_objs as go
import plotly.tools as tls
import colorlover as cl

#####################################################
#####################################################

                ##### STOCK X #####
                     # DATA #
                #### CHALLENGE ####
            
#####################################################
#####################################################

path = 'https://s3.amazonaws.com/stockx-sneaker-analysis/wp-content/uploads/2019/02/StockX-Data-Contest-2019.xlsx'

df = pd.read_excel(path, sheet_name=1)

df.head()

Order Date   Brand                                   Sneaker Name  \
0 2017-09-01   Yeezy           Adidas-Yeezy-Boost-350-Low-V2-Beluga   
1 2017-09-01   Yeezy    Adidas-Yeezy-Boost-350-V2-Core-Black-Copper   
2 2017-09-01   Yeezy     Adidas-Yeezy-Boost-350-V2-Core-Black-Green   
3 2017-09-01   Yeezy       Adidas-Yeezy-Boost-350-V2-Core-Black-Red   
4 2017-09-01   Yeezy  Adidas-Yeezy-Boost-350-V2-Core-Black-Red-2017   

   Sale Price  Retail Price Release Date  Shoe Size  Buyer Region  
0      1097.0           220   2016-09-24       11.0    California  
1       685.0           220   2016-11-23       11.0    California  
2       690.0           220   2016-11-23       11.0    California  
3      1075.0           220   2016-11-23       11.5      Kentucky  
4       828.0           220   2017-02-11       11.0  Rhode Island

In [9]:
type(df)

pandas.core.frame.DataFrame

In [11]:
df.corr()

Sale Price  Retail Price  Shoe Size
Sale Price      1.000000     -0.361549   0.082795
Retail Price   -0.361549      1.000000  -0.052798
Shoe Size       0.082795     -0.052798   1.000000

## From the blog post 
The data we’re giving you consists of a random sample of all Off-White x Nike and Yeezy 350 sales from between 9/1/2017 (the month that Off-White first debuted “The Ten” collection) and the present. There are 99,956 total sales in the data set; 27,794 Off-White sales, and 72,162 Yeezy sales. The sample consists of U.S. sales only.

To create this sample, we took a random, fixed percentage of StockX sales (X%) for each colorway, on each day, since September 2017. So, for each day the Off-White Jordan 1 was on the market, we randomly selected X% of its sale from each day. (It’s not important to know what X is; all that matters is that it’s a random sample, and that the same fixed X% of sales was selected from every day, for every sneaker).

We’ve included 8 variables for you to work with: Order Date, Brand, Sneaker Name, Sale Price ($), Retail Price ($), Release Date, Shoe Size, and Buyer State (the U.S. state the buyer shipped to). You can use whatever variables you want in the analysis; you can use 1 variable, or you can use all 8. And remember, every row in the spreadsheet represents an individual StockX sale. There are no averages or order counts; this is just a random sample of daily sales data

In [25]:
## Creating data frames for yeezys and offwhite shoes

off_white = df.copy()
off_white = off_white[off_white['Brand'].isin(['Off-White'])]

yeezy = df.copy()
yeezy = yeezy[yeezy['Brand'].isin([' Yeezy'])]

print('Shape of Off-White dataframe : {}'.format(off_white.shape))
print('Shape of Yeezy dataframe : {}'.format(yeezy.shape))

Shape of Off-White dataframe : (27794, 8)
Shape of Yeezy dataframe : (72162, 8)


In [27]:
#lmao this dataframe is only yeezys and offwhite i dig it
df['Brand'].unique()

array([' Yeezy', 'Off-White'], dtype=object)

In [28]:
yeezy['Sneaker Name'].unique()

array(['Adidas-Yeezy-Boost-350-Low-V2-Beluga',
       'Adidas-Yeezy-Boost-350-V2-Core-Black-Copper',
       'Adidas-Yeezy-Boost-350-V2-Core-Black-Green',
       'Adidas-Yeezy-Boost-350-V2-Core-Black-Red',
       'Adidas-Yeezy-Boost-350-V2-Core-Black-Red-2017',
       'Adidas-Yeezy-Boost-350-V2-Core-Black-White',
       'Adidas-Yeezy-Boost-350-V2-Cream-White',
       'Adidas-Yeezy-Boost-350-V2-Zebra',
       'Adidas-Yeezy-Boost-350-Low-Moonrock',
       'Adidas-Yeezy-Boost-350-Low-Pirate-Black-2016',
       'Adidas-Yeezy-Boost-350-Low-Oxford-Tan',
       'Adidas-Yeezy-Boost-350-Low-Turtledove',
       'Adidas-Yeezy-Boost-350-Low-Pirate-Black-2015',
       'Adidas-Yeezy-Boost-350-V2-Semi-Frozen-Yellow',
       'Adidas-Yeezy-Boost-350-V2-Beluga-2pt0',
       'Adidas-Yeezy-Boost-350-V2-Blue-Tint',
       'adidas-Yeezy-Boost-350-V2-Butter',
       'Adidas-Yeezy-Boost-350-V2-Sesame',
       'adidas-Yeezy-Boost-350-V2-Static',
       'adidas-Yeezy-Boost-350-V2-Static-Reflective'], dtype=objec

In [29]:
off_white['Sneaker Name'].unique()

array(['Nike-Air-Max-90-Off-White', 'Nike-Air-Presto-Off-White',
       'Nike-Air-VaporMax-Off-White',
       'Air-Jordan-1-Retro-High-Off-White-Chicago',
       'Nike-Blazer-Mid-Off-White', 'Nike-Air-Force-1-Low-Off-White',
       'Nike-Air-Max-97-Off-White',
       'Nike-Air-Force-1-Low-Virgil-Abloh-Off-White-AF100',
       'Nike-React-Hyperdunk-2017-Flyknit-Off-White',
       'Nike-Zoom-Fly-Off-White', 'Nike-Air-VaporMax-Off-White-2018',
       'Air-Jordan-1-Retro-High-Off-White-White',
       'Nike-Air-VaporMax-Off-White-Black',
       'Air-Jordan-1-Retro-High-Off-White-University-Blue',
       'Nike-Air-Presto-Off-White-Black-2018',
       'Nike-Air-Presto-Off-White-White-2018',
       'Nike-Zoom-Fly-Mercurial-Off-White-Black',
       'Nike-Zoom-Fly-Mercurial-Off-White-Total-Orange',
       'Nike-Air-Max-97-Off-White-Elemental-Rose-Queen',
       'Nike-Blazer-Mid-Off-White-All-Hallows-Eve',
       'Nike-Blazer-Mid-Off-White-Grim-Reaper',
       'Nike-Blazer-Mid-Off-White-Wolf-Grey

# Ideas for graphs/charts/trends to explore

- Region/Map of where the most buyers are for yeezys vs. offwhite sneakers
- and what does the this trend ^ explain about the economy in that state or how much disposable income people have to be spending $1,000 on a pair of sneakers LOL
- Another trend: what sneakers are more popular? Kind of obvious because there are almost 3X as many yeezys sold than offwhite in the entire dataset
- Another good one would be to look at the difference between resale prices for the shoes?
- Also how the resale prices change over time
- Do the shoes appreciate? or do they not?

## After brainstorming:

- Has 'The Ten' Appreciated or Depreciated in the last year?
- Have Yeezys gone downhill?

In [32]:
the_ten = off_white.copy()

the_ten = the_ten[the_ten['Sneaker Name'].isin([
    'Air-Jordan-1-Retro-High-Off-White-Chicago',
    'Nike-Blazer-Mid-Off-White',
    'Nike-Air-VaporMax-Off-White',
    'Nike-Air-Presto-Off-White',
    'Nike-Air-Max-90-Off-White',
    'Nike-Air-Force-1-Low-Off-White',
    'Nike-Air-Max-97-Off-White',
    'Nike-Air-Force-1-Low-Virgil-Abloh-Off-White-AF100',
    'Nike-React-Hyperdunk-2017-Flyknit-Off-White',
    'Nike-Zoom-Fly-Off-White' 
])]
the_ten.head()

Order Date      Brand                 Sneaker Name  Sale Price  \
128 2017-09-07  Off-White    Nike-Air-Max-90-Off-White      1600.0   
129 2017-09-07  Off-White    Nike-Air-Max-90-Off-White      1090.0   
130 2017-09-07  Off-White    Nike-Air-Presto-Off-White      1344.0   
131 2017-09-07  Off-White    Nike-Air-Presto-Off-White      1325.0   
132 2017-09-07  Off-White  Nike-Air-VaporMax-Off-White      1800.0   

     Retail Price Release Date  Shoe Size   Buyer Region  
128           160   2017-09-09        8.0     California  
129           160   2017-09-09       11.5       New York  
130           160   2017-09-09       10.0       New York  
131           160   2017-09-09       10.0  Massachusetts  
132           250   2017-09-09       12.0       Kentucky

In [35]:
the_ten.tail()

Order Date      Brand                                 Sneaker Name  \
99510 2019-02-12  Off-White                      Nike-Zoom-Fly-Off-White   
99737 2019-02-13  Off-White    Air-Jordan-1-Retro-High-Off-White-Chicago   
99742 2019-02-13  Off-White               Nike-Air-Force-1-Low-Off-White   
99842 2019-02-13  Off-White                    Nike-Air-Presto-Off-White   
99858 2019-02-13  Off-White  Nike-React-Hyperdunk-2017-Flyknit-Off-White   

       Sale Price  Retail Price Release Date  Shoe Size Buyer Region  
99510       821.0           170   2017-11-01       10.0       Oregon  
99737      2550.0           190   2017-09-09       11.0       Oregon  
99742      1315.0           170   2017-11-01       11.0   California  
99842      1700.0           160   2017-09-09       10.0   California  
99858       767.0           200   2017-11-01       11.0   California

# Apparently highsnobiety already made an article on the appreciation of Off White sneakers, so i'm going to do something different lmao


## here's a link to it


https://www.highsnobiety.com/p/off-white-nike-resell-price-analysis/

